RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -0,0 +1,127 @@
+package com.rfs.common;
+
+import java.net.HttpURLConnection;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import reactor.core.publisher.Mono;
+
+public class OpenSearchClient {
+    private static final Logger logger = LogManager.getLogger(OpenSearchClient.class);
+
+    public static class BulkResponse extends RestClient.Response {
+        public BulkResponse(int responseCode, String responseBody, String responseMessage) {
+            super(responseCode, responseBody, responseMessage);
+        }
+
+        public boolean hasBadStatusCode() {
+            return !(code == HttpURLConnection.HTTP_OK || code == HttpURLConnection.HTTP_CREATED);
+        }
+
+        public boolean hasFailedOperations() {
+            return body.contains("\"errors\":true");
```

Comment:
Leave a TODO to strengthen this so that you're looking in the right place or a comment as to why it works?  You might already be immune to checking within a document if the documents are already json encoded/escaped.

Reply:
Left a comment; hope it's satisfactory.